### PR TITLE
feat: support tree shaking of individual wallets

### PR DESCRIPTION
**Pull request by markdalgleish**
In order to reduce bundle size, you can now select the individual wallets you want to import into your application.

Note that since wallets are no longer namespaced via the `wallet` object, all wallets now have a `Wallet` suffix.

```diff
-import { connectorsForWallets, wallet } from '@rainbow-me/rainbowkit';
+import { connectorsForWallets } from '@rainbow-me/rainbowkit';
+import {
+  injectedWallet,
+  rainbowWallet,
+  metaMaskWallet,
+  coinbaseWallet,
+  walletConnectWallet,
+} from '@rainbow-me/rainbowkit/wallets';

const wallets = [
-  wallet.injected({ chains }),
-  wallet.rainbow({ chains }),
-  wallet.metaMask({ chains }),
-  wallet.coinbase({ chains, appName: 'My App' }),
-  wallet.walletConnect({ chains }),
+  injectedWallet({ chains }),
+  rainbowWallet({ chains }),
+  metaMaskWallet({ chains }),
+  coinbaseWallet({ chains, appName: 'My App' }),
+  walletConnectWallet({ chains }),
];
```

Since this is a breaking change, I also took the opportunity to remove the Steakwallet backwards compatibility layer which we had previously marked as deprecated.

### CODE REVIEW

File: packages/rainbowkit/wallets/package.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "main": "../dist/wallets/walletConnectors/index.js"
```
Comment (markdalgleish) [inline packages/rainbowkit/wallets/package.json:2]: Similar to our `styles.css` alias, this is here mainly as backwards compatibility for Jest.

File: site/data/docs/custom-wallet-list.mdx [hunk 2]
```diff
@@ -146,51 +151,51 @@ wallet.ledger(options: {
...
 
-wallet.metaMask(options: {
+metaMaskWallet(options: {
   chains: Chain[];
   shimDisconnect?: boolean;
 });
 ```
 
-#### Rainbow
+#### Omni
```
Comment (markdalgleish) [inline site/data/docs/custom-wallet-list.mdx:162]: I noticed that the docs hadn't been updated with the Steakwallet to Omni rename, so I updated them as part of this.

File: site/data/docs/installation.mdx [hunk 3]
```diff
@@ -125,7 +125,19 @@ If your bundler doesn't provide Node polyfills (e.g. [Vite](https://vitejs.dev))
 
 ##### Remix
 
-When using [Remix](https://remix.run), RainbowKit must be added to your list of [server dependencies](https://remix.run/docs/en/v1/pages/gotchas#importing-esm-packages) since it's published as an ESM package.
+When using [Remix](https://remix.run), all RainbowKit package entry points must be added to your list of [server dependencies](https://remix.run/docs/en/v1/pages/gotchas#importing-esm-packages) in your Remix config since they're published as ESM packages.
```
Comment (markdalgleish) [inline site/data/docs/installation.mdx:128]: I updated this copy to cover the fact that RainbowKit is now a suite of multiple packages and entry points. This was a bit of a gotcha, almost missed this one.

**Review by peduarte — APPROVED** [review]
👌

**Review by jxom — APPROVED** [review]
Really solid!

**Pull request merged**